test/jdk/javax/swing/JFileChooser/6698013/bug6698013.java

```diff
@@ -43,9 +44,11 @@
     final static VirtualFile subdirFile = new VirtualFile("testdir/subdir/subtest.txt", false);
 
     public static void main(String[] args) {
-        JFileChooser chooser = new JFileChooser(new VirtualFileSystemView());
-        chooser.setCurrentDirectory(root);
-        chooser.showOpenDialog(null);
+        try {
+            SwingUtilities.invokeAndWait(() -> new bug6698013().init());
+        } catch (InvocationTargetException | InterruptedException e) {
+           e.printStackTrace();
```

Comment:
The test should, probably, fail in case of the exception.

Reply:
Yes, it's safe to declare `main` with `throws Exception`; `main` is for convenience to run the test quickly on the command line.

The test harness does not use `main`, it runs the test as applet.